File: LoopBack/src/controllers/student.controller.ts
import {
  repository,
} from '@loopback/repository';
import {
  param,
  get,
} from '@loopback/rest';
import {Student} from '../models';
import {StudentRepository} from '../repositories';

export class StudentController {
  constructor(
    @repository(StudentRepository)
    public studentRepository : StudentRepository,
  ) {}

  @get('/students/add')
  async addStudent(
    @param.query.string('firstName') firstName: string,
    @param.query.string('lastName') lastName: string,
    @param.query.string('group') group: string,
  ): Promise<Student> {
    return await this.studentRepository.create({
      "firstName": firstName,
      "lastName": lastName,
      "group": group,
    });
  }

  @get('/students/all')
  async getAll(): Promise<Student[]> {
    return await this.studentRepository.find();
  }

  @get('/students/{id}')
  async getStudent(@param.path.string('id') id: string): Promise<Student> {
    return await this.studentRepository.findById(id);
  }
}

File: Spring Boot/demo/src/main/java/controller/StudentController.java
package controller;

import java.util.Hashtable;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import model.Group;
import model.Student;
import service.GroupService;
import service.StudentService;

@RestController
@RequestMapping("/students")
public class StudentController {
	@Autowired
	StudentService ss;
	
	@Autowired
	GroupService gs;
	
	@RequestMapping("add")
	public Student addStudent(
			@RequestParam(name="id") String id,
			@RequestParam(name="firstName") String firstName,
			@RequestParam(name="lastName") String lastName,
			@RequestParam(name="group") String group
		) {
		
		Group g = gs.getGroup(group);
		ss.addStudent(id, firstName, lastName, g.getId());
		return ss.getStudent(id);
	}
	
	@RequestMapping("/all")
	public Hashtable<String, Student> getAll() {
		return ss.getAll();
	}
	
	@RequestMapping("{id}")
	public Student getStudent(@PathVariable("id") String id) {
		return ss.getStudent(id);
	}
}
